hazelcast/src/main/java/com/hazelcast/client/impl/protocol/codec/builtin/CustomTypeFactory.java

```diff
@@ -248,7 +248,18 @@ public static FieldDescriptor createFieldDescriptor(@Nonnull String fieldName, i
     }
 
     public static Schema createSchema(String typeName, List<FieldDescriptor> fields) {
-        return new Schema(typeName, fields);
+        String[] metadata = typeName.split("\0", 2);
+        if (metadata.length == 1) {
+            return new Schema(typeName, fields);
+        } else {
+            try {
+                Class<Schema> schemaClass = (Class<Schema>) CustomTypeFactory.class.getClassLoader()
```

Comment:
loading class with name coming from the user and invoking its constructor might be a security risk